app/org/sagebionetworks/bridge/models/schedules/SchedulerFactory.java

```diff
@@ -0,0 +1,18 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import org.sagebionetworks.bridge.validators.ScheduleValidator;
+import org.sagebionetworks.bridge.validators.Validate;
+
+public class SchedulerFactory {
```

Comment:
It'll be called at request time when a study's schedule plans are retrieved, the schedule for the user is retrieved, and then the scheduler is called to retrieve the tasks. These will be compared against tasks already created for the user, and saved if they are new (extending forwarding time from the last request), before the list of tasks are returned to the user.